### PR TITLE
update: windows 0.51.1, shroud 0.3.1

### PR DESCRIPTION
**Pull request by tobii-dev**
Right now, 2023-09-21, the windows crate is still at 0.48.0, but with this PR, the egui-d3d9 crate can be ready for windows 0.51.1

Also updates the shroud crate from 0.2.2 to 0.3.1

I have not tested this properly.

### CODE REVIEW

File: example/src/lib.rs [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-use retour::static_detour;
 use std::{
     intrinsics::transmute,
     sync::{Arc, Once},
     time::Duration,
 };
 use windows::{
-    core::{HRESULT, PCSTR},
-    s,
+    core::{HRESULT, PCSTR, s},
```
Comment (tobii-dev) [inline example/src/lib.rs]: https://microsoft.github.io/windows-docs-rs/doc/windows/core/macro.s.html

File: egui-d3d9/src/state.rs [hunk 2]
```diff
@@ -91,7 +89,8 @@ impl Drop for DxState {
         // restore the previous state
         unsafe {
             expect!(
-                self.dev.SetTransform(D3DTS_WORLD, &self.original_world),
+                self.dev
+                    .SetTransform(D3DTRANSFORMSTATETYPE(0 + 256), &self.original_world),
```
Comment (tobii-dev) [inline egui-d3d9/src/state.rs:93]: https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dts-worldmatrix

For some reason the D3DTS_WORLD macro is gone on the new version?

https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dts-world

File: egui-d3d9/src/state.rs [hunk 3]
```diff
@@ -54,7 +51,8 @@ impl DxState {
             let mut original_proj: Matrix4x4 = Default::default();
 
             expect!(
-                dev.GetTransform(D3DTS_WORLD, &mut original_world),
+                // https://github.com/apitrace/dxsdk/blob/d964b66467aaa734edbc24326da8119f5f063dd3/Include/d3d9types.h#L333C35-L333C56
+                dev.GetTransform(D3DTRANSFORMSTATETYPE(0 + 256), &mut original_world),
```
Comment (tobii-dev) [inline egui-d3d9/src/state.rs:55]: https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dts-worldmatrix

For some reason the D3DTS_WORLD macro is gone on the new version?

https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dts-world

**Review by tobii-dev — COMMENTED** [review]
Untested so far!

**Comment by unknowntrojan** [comment]
Thanks for your contribution! GitHub's notification got marked as spam, sorry for the late response.

**Pull request merged**